xcode/Subconscious/Shared/Components/AddressBook/AddressBookView.swift

```diff
@@ -0,0 +1,138 @@
+    ]
+    
+    var myDid = "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"
+    
+    func delete(at offsets: IndexSet) {
+        following.remove(atOffsets: offsets)
+    }
+    
+    func filteredFollowing() -> [AddressBookEntry] {
+        return following.filter { user in
+            if (searchText.count > 0) {
+                return user.petname.lowercased().range(of: searchText.lowercased()) != nil
+            } else {
+                return true
+            }
+        }
+    }
+
+    var body: some View {
+        NavigationStack {
+            Form {
+                if (following.count == 0) {
+                    Text("No friends?")
+                } else {
+                    Section(header: Text("Friends")) {
+                        List {
+                            ForEach(filteredFollowing(), id: \.did) { user in
+                                AddressBookEntryView(pfp: user.pfp, petname: user.petname,
+                                                     did: user.did)
+                            }
+                            .onDelete(perform: delete)
+                        }
+                        .searchable(text: $searchText)
+                    }
+                }
+                Section(header: Text("My Details")) {
+                   DidQrCodeView(did: myDid)
+                        .frame(maxWidth: .infinity, maxHeight: 200)
+                    
+                    VStack {
+                        
+                        HStack{
+                        Text(myDid)
+                                .foregroundColor(.secondary)
+                            Spacer()
+                            Button(action: {
+                                // TODO: actually copy it!
+                                didCopy = true
+                            }, label: {
+                                if !didCopy {
+                                    HStack {
+                                        Image(systemName: "doc.on.doc")
+                                    }
+                                    .transition(
+                                        .asymmetric(
+                                            insertion: .identity,
+                                            removal: .move(
+                                                edge: .top
+                                            ).combined(
+                                                with: .opacity
+                                            )
+                                        )
+                                    )
+                                } else {
+                                    HStack {
+                                        Image(systemName: "checkmark.circle")
+                                    }
+                                    .transition(.opacity)
+                                }
```

Comment:
Trimmed down (icon only) version of the copy component from the First Run experience